### ISSUE TITLE
Recover panics from within cmds goroutines

### ISSUE THREAD

**Issue by brooks-connor-a** [charmbracelet/bubbletea#234]
Panics are handled by recovering the panic and gracefully tearing down the Program. However goroutines started within the Program are outside the scope of the recover, and as such can end in the program terminating without properly tearing down. This can result in the terminal getting stuck in a non-interactive state until it is reset.

Here's a minimal reproduction:
```
package main
​
import ( 
    tea "github.com/charmbracelet/bubbletea"
)
​
type model struct {}

func (m model) Init() tea.Cmd {
    return nil
}

func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) { 
    return m, func() tea.Msg { 
        if true { 
            panic("")
        } 
        return tea.KeyEnter
    } 
}
​
func (m model) View() string { 
    return ""
}
​
func main() { 
    p := tea.NewProgram(model{})
    p.Start()
}
```

**Comment by decentral1se** [comment]
Thanks for reporting @brooks-connor-a, ran into this also.

**Comment by joerdav** [comment]
I took a look at this, it seems like the best way to solve was to recover the panic and write to the `errs` channel. Happy for some feedback if it's not in the spirit of how bubbletea works.

**Comment by meowgorithm** [comment]
Hi! Just a note that this is now fixed and available in [Bubble Tea v1.1.1](https://github.com/charmbracelet/bubbletea/releases/tag/v1.1.1). You can use `tea.Kill` to reset the terminal from panics externally.

```go
p := tea.NewProgram(model{})

go func() {
	time.Sleep(3 * time.Second)
	defer p.Kill()
	panic("Oh no!")
}()
	
if _, err := p.Run() {
	log.Fatal(err)
}
```
